default/scripting/buildings/colonies/SP_BANFORO.focs.txt

```diff
@@ -45,18 +45,21 @@ BuildingType
         Population high = 0
         Not Planet environment = Uninhabitable species = "SP_BANFORO"
         Not Contains Building name = "BLD_COL_BANFORO"
-        ResourceSupplyConnected empire = Source.Owner condition = And [
-            Planet
-            OwnedBy empire = Source.Owner
-            Or [
-                And [
-                    Species name = "SP_BANFORO"
-                    Population low = [[MIN_RECOLONIZING_SIZE]]
-                    Happiness low = 5
-                ]
-                And [
-                    HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
-                    Contains Building name = "BLD_XENORESURRECTION_LAB"
+        Or [
+            OwnerHasTech name = "TECH_COL_BANFORO"
+            ResourceSupplyConnected empire = Source.Owner condition = And [
+                Planet
+                OwnedBy empire = Source.Owner
+                Or [
+                    And [
+                        Species name = "SP_BANFORO"
+                        Population low = [[MIN_RECOLONIZING_SIZE]]
+                        Happiness low = 5
+                    ]
+                    And [
+                        HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
+                        Contains Building name = "BLD_XENORESURRECTION_LAB"
+                    ]
```

Comment:
Correct me if I'm wrong, but shouldn't the `OwnerHasTech name = "TECH_COL_BANFORO"` condition be together with the
```
HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
Contains Building name = "BLD_XENORESURRECTION_LAB"
```
block? Because if I read the implementation above correctly, having the Banforo colony tech alone would allow you to build a Banforo colony (the outer `OR` would be satisfied, and the `ResourceSupplyConnected` condition become irrelevant). Which I don't think is what you want to achieve.

So IMO it should look like this:
```
ResourceSupplyConnected empire = Source.Owner condition = And [
  Planet
  OwnedBy empire = Source.Owner
  Or [
    And [
      Species name = "SP_BANFORO"
      Population low = [[MIN_RECOLONIZING_SIZE]]
      Happiness low = 5
    ]
    And [
      OwnerHasTech name = "TECH_COL_BANFORO"
      HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
      Contains Building name = "BLD_XENORESURRECTION_LAB"
    ]
  ]
]
```
Which should enable you to build a Banforo colony if either:

* you own an already established Banforo colony that meets the pop and happiness tresholds, and is supply connected to the planet on which you want to build the new colony.
* have the Banforo colony tech, and own a colony with the Banforo special and the Xenoresurrection Lab building which is supply connected to the planet on which you want to build the new colony.